Sources/ClientRuntime/Interceptor/AnyInterceptor.swift

```diff
@@ -0,0 +1,139 @@
+    private var readBeforeSerialization: InterceptorFn
+    private var readAfterSerialization: InterceptorFn
+    private var modifyBeforeRetryLoop: MutatingInterceptorFn
+    private var readBeforeAttempt: InterceptorFn
+    private var modifyBeforeSigning: MutatingInterceptorFn
+    private var readBeforeSigning: InterceptorFn
+    private var readAfterSigning: InterceptorFn
+    private var modifyBeforeTransmit: MutatingInterceptorFn
+    private var readBeforeTransmit: InterceptorFn
+    private var readAfterTransmit: InterceptorFn
+    private var modifyBeforeDeserialization: MutatingInterceptorFn
+    private var readBeforeDeserialization: InterceptorFn
+    private var readAfterDeserialization: InterceptorFn
+    private var modifyBeforeAttemptCompletion: MutatingInterceptorFn
+    private var readAfterAttempt: InterceptorFn
+    private var modifyBeforeCompletion: MutatingInterceptorFn
+    private var readAfterExecution: InterceptorFn
+
+    internal init<I: Interceptor>(interceptor: I)
+    where
+        I.RequestType == RequestType,
+        I.ResponseType == ResponseType,
+        I.AttributesType == AttributesType {
+        self.readBeforeExecution = interceptor.readBeforeExecution(context:)
+        self.modifyBeforeSerialization = interceptor.modifyBeforeSerialization(context:)
+        self.readBeforeSerialization = interceptor.readBeforeSerialization(context:)
+        self.readAfterSerialization = interceptor.readAfterSerialization(context:)
+        self.modifyBeforeRetryLoop = interceptor.modifyBeforeRetryLoop(context:)
+        self.readBeforeAttempt = interceptor.readBeforeAttempt(context:)
+        self.modifyBeforeSigning = interceptor.modifyBeforeSigning(context:)
+        self.readBeforeSigning = interceptor.readBeforeSigning(context:)
+        self.readAfterSigning = interceptor.readAfterSigning(context:)
+        self.modifyBeforeTransmit = interceptor.modifyBeforeTransmit(context:)
+        self.readBeforeTransmit = interceptor.readBeforeTransmit(context:)
+        self.readAfterTransmit = interceptor.readAfterTransmit(context:)
+        self.modifyBeforeDeserialization = interceptor.modifyBeforeDeserialization(context:)
+        self.readBeforeDeserialization = interceptor.readBeforeDeserialization(context:)
+        self.readAfterDeserialization = interceptor.readAfterDeserialization(context:)
+        self.modifyBeforeAttemptCompletion = interceptor.modifyBeforeAttemptCompletion(context:)
+        self.readAfterAttempt = interceptor.readAfterAttempt(context:)
+        self.modifyBeforeCompletion = interceptor.modifyBeforeCompletion(context:)
+        self.readAfterExecution = interceptor.readAfterExecution(context:)
+    }
+
+    internal func readBeforeExecution(context: InterceptorContextType) async throws {
+        try await self.readBeforeExecution(context)
+    }
+
+    internal func modifyBeforeSerialization(context: inout InterceptorContextType) async throws {
```

Comment:
What is the purpose for making the `context` param `inout` in this & the other interceptor hooks?

`InterceptorContextType` is a typealias for `DefaultInterceptorContext`.  `DefaultInterceptorContext` is a class, which is a Swift reference type.  

The `inout` modifier means that the reference to `context` can be modified by the called method to point to a different `context`, which isn't what I think you want.  (`inout` would be necessary if the passed context was a Swift value type such as `struct` and you wanted to change the struct's properties, but that is not the case here.)

Reply:
No purpose. Working on the [orchestrator PR](https://github.com/smithy-lang/smithy-swift/pull/694) I realized it and just made `InterceptorContext` `AnyObject`, so there's no more `inout`s anywhere. Need to update this PR